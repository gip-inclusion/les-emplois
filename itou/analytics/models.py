import uuid

from django.db import models
from django.utils import timezone

from itou.common_apps.address.departments import DEPARTMENTS
from itou.users.enums import UserKind


class DatumCode(models.TextChoices):
    # Employee record - Base
    EMPLOYEE_RECORD_COUNT = "ER-001", "FS totales"
    EMPLOYEE_RECORD_DELETED = "ER-002", "FS (probablement) supprimées"
    # Employee record - Lifecycle
    EMPLOYEE_RECORD_PROCESSED_AT_FIRST_EXCHANGE = "ER-101", "FS intégrées (0000) au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_AT_FIRST_EXCHANGE = "ER-102", "FS avec une erreur au premier retour"
    EMPLOYEE_RECORD_WITH_ERROR_3436_AT_FIRST_EXCHANGE = "ER-102-3436", "FS avec une erreur 3436 au premier retour"
    EMPLOYEE_RECORD_WITH_AT_LEAST_ONE_ERROR = "ER-103", "FS ayant eu au moins un retour en erreur"
    # Approval - Base
    APPROVAL_COUNT = "AP-001", "PASS IAE total"
    APPROVAL_CANCELLED = "AP-002", "PASS IAE annulés"
    # Approval - PE notification cycle
    APPROVAL_PE_NOTIFY_SUCCESS = "AP-101", "PASS IAE synchronisés avec succès avec pole emploi"
    APPROVAL_PE_NOTIFY_PENDING = "AP-102", "PASS IAE en attente de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_ERROR = "AP-103", "PASS IAE en erreur de synchronisation avec pole emploi"
    APPROVAL_PE_NOTIFY_READY = "AP-104", "PASS IAE prêts à être synchronisés avec pole emploi"
    # Users
    USER_COUNT = "US-001", "Nombre d'utilisateurs"
    USER_JOB_SEEKER_COUNT = "US-011", "Nombre de demandeurs d'emploi"
    USER_PRESCRIBER_COUNT = "US-012", "Nombre de prescripteurs"
    USER_EMPLOYER_COUNT = "US-013", "Nombre d'employeurs"
    USER_LABOR_INSPECTOR_COUNT = "US-014", "Nombre d'inspecteurs du travail"
    USER_ITOU_STAFF_COUNT = "US-015", "Nombre d'administrateurs"
    # API usage
    API_TOTAL_CALLS = "API-001", "API : total d'appels reçus"
    API_TOTAL_UV = "API-002", "API : total de visiteurs uniques"
    API_TOTAL_CALLS_CANDIDATS = "API-003", "API candidats : total d'appels reçus"
    API_TOTAL_UV_CANDIDATS = "API-004", "API candidats : total de visiteurs uniques"
    API_TOTAL_CALLS_GEIQ = "API-005", "API GEIQ : total d'appels reçus"
    API_TOTAL_UV_GEIQ = "API-006", "API GEIQ : total de visiteurs uniques"
    API_TOTAL_CALLS_ER = "API-007", "API FS : total d'appels reçus"
    API_TOTAL_UV_ER = "API-008", "API FS : total de visiteurs uniques"
    API_TOTAL_CALLS_MARCHE = "API-009", "API Le marché : total d'appels reçus"
    API_TOTAL_CALLS_SIAES = "API-010", "API siaes : total d'appels reçus"
    API_TOTAL_UV_SIAES = "API-011", "API siaes : total de visiteurs uniques (par adresse IP)"
    API_TOTAL_CALLS_STRUCTURES = "API-012", "API structures : total d'appels reçus"


class Datum(models.Model):
    """Store an aggregated `value` of the `code` data point for the specified `bucket`."""

    id = models.UUIDField(primary_key=True, default=uuid.uuid4)

    code = models.TextField(choices=DatumCode.choices)
    bucket = models.TextField()
    value = models.IntegerField()  # Integer offers the best balance between range, storage size, and performance

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "data"
        unique_together = ["code", "bucket"]
        indexes = [models.Index(fields=["measured_at", "code"])]


class StatsDashboardVisit(models.Model):
    dashboard_id = models.IntegerField(verbose_name="ID tableau de bord Metabase")
    dashboard_name = models.TextField(verbose_name="nom de la vue du tableau de bord")
    department = models.CharField(verbose_name="département", choices=DEPARTMENTS.items(), max_length=3, null=True)
    region = models.TextField(verbose_name="région", null=True)
    current_company_id = models.IntegerField(verbose_name="ID entreprise courante", null=True)
    current_prescriber_organization_id = models.IntegerField(
        verbose_name="ID organisation prescriptrice courante", null=True
    )
    current_institution_id = models.IntegerField(verbose_name="ID institution courante", null=True)
    user_kind = models.TextField(verbose_name="type d'utilisateur", choices=UserKind.choices)
    user_id = models.IntegerField(verbose_name="ID utilisateur")

    measured_at = models.DateTimeField(default=timezone.now)  # Not using auto_now_add=True to allow overrides

    class Meta:
        verbose_name_plural = "visite de tableau de bord"
