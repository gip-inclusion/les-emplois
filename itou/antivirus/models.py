from django.db import models
from django.utils.text import capfirst

from itou.files.models import File


class Scan(models.Model):
    file = models.OneToOneField(File, on_delete=models.CASCADE)
    clamav_signature = models.TextField()
    clamav_completed_at = models.DateTimeField(null=True, verbose_name="analyse ClamAV le")
    clamav_infected = models.BooleanField(null=True, verbose_name="fichier infect√© selon ClamAV")
    comment = models.TextField(blank=True, verbose_name="commentaire")

    class Meta:
        verbose_name = "analyse antivirus"
        verbose_name_plural = "analyses antivirus"

    def __str__(self):
        text = f"{capfirst(self._meta.verbose_name)} {self.file_id}"
        if self.clamav_infected:
            text = f"[VIRUS] {text}"
        return text
