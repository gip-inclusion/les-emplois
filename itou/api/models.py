import secrets

from django.db import models
from django.utils import timezone

from itou.companies.models import Company


def _generate_key():
    return secrets.token_urlsafe()


class CompanyToken(models.Model):
    key = models.CharField(default=_generate_key, unique=True, editable=False)
    label = models.CharField(verbose_name="m√©mo permettant d'identifier l'usage du jeton", max_length=60, unique=True)
    created_at = models.DateTimeField(default=timezone.now)
    companies = models.ManyToManyField(Company, related_name="api_tokens")

    class Meta:
        verbose_name = "jeton d'API entreprise"
        verbose_name_plural = "jetons d'API entreprise"
