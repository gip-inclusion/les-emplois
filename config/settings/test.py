import os


ITOU_ENVIRONMENT = "DEV"
os.environ["ITOU_ENVIRONMENT"] = ITOU_ENVIRONMENT

from .base import *  # noqa: E402,F403


SECRET_KEY = "foobar"
ALLOWED_HOSTS = []

# We *want* to do the same `collectstatic` on the CI than on PROD to catch errors early,
# but we don't want to do it when running the test suite locally for performance reasons.
if not os.getenv("CI", False):
    # `ManifestStaticFilesStorage` (used in base settings) requires `collectstatic` to be run.
    STORAGES["staticfiles"]["BACKEND"] = "django.contrib.staticfiles.storage.StaticFilesStorage"  # noqa: F405

ASP_ITOU_PREFIX = "XXXXX"  # same as in our fixtures
ITOU_PROTOCOL = "http"
ITOU_FQDN = "localhost:8000"

DATABASES["default"]["HOST"] = os.getenv("PGHOST", "127.0.0.1")  # noqa: F405
DATABASES["default"]["PORT"] = os.getenv("PGPORT", "5432")  # noqa: F405
DATABASES["default"]["NAME"] = os.getenv("PGDATABASE", "itou")  # noqa: F405
DATABASES["default"]["USER"] = os.getenv("PGUSER", "postgres")  # noqa: F405
DATABASES["default"]["PASSWORD"] = os.getenv("PGPASSWORD", "password")  # noqa: F405

MAILJET_API_KEY_PRINCIPAL = "API_MAILJET_KEY_PRINCIPAL"
MAILJET_SECRET_KEY_PRINCIPAL = "API_MAILJET_SECRET_PRINCIPAL"

AWS_S3_ENDPOINT_URL = f"http://{os.getenv('CELLAR_ADDON_HOST', 'localhost:9000')}/"
AWS_S3_ACCESS_KEY_ID = "minioadmin"
AWS_S3_SECRET_ACCESS_KEY = "minioadmin"
AWS_STORAGE_BUCKET_NAME = "tests"
