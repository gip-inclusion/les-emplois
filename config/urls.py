from django.conf import settings
from django.contrib import admin
from django.urls import include, path, re_path, register_converter
from django.views.generic import TemplateView

from itou.utils.urls import SiretConverter
from itou.www.dashboard import views as dashboard_views
from itou.www.login import views as login_views
from itou.www.signup import views as signup_views


register_converter(SiretConverter, "siret")

urlpatterns = [
    path("admin/", admin.site.urls),
    # --------------------------------------------------------------------------------------
    # allauth URLs. Order is important because some URLs are overriden.
    # --------------------------------------------------------------------------------------
    # Override allauth `account_signup` URL.
    # /accounts/signup/ <=> account_signup
    # We don't want any user to be able to signup using the default allauth `signup` url
    # because we have multiple specific signup processes for different kind of users.
    re_path(r"^accounts/signup/$", signup_views.signup),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_login` URL.
    # /accounts/login/ <=> account_login
    # We override this view because the login page should look slightly differently
    # for job seekers, prescribers and employers.
    # Also, PEAMU is only available for job seekers.
    re_path(r"^accounts/login/$", login_views.login),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_change_password` URL.
    # /accounts/password/change/ <=> account_change_password
    # https://github.com/pennersr/django-allauth/issues/468
    re_path(r"^accounts/password/change/$", dashboard_views.password_change),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_reset_password` URL.
    # Avoid user enumeration via password reset page.
    re_path(r"^accounts/password/reset/$", signup_views.ItouPasswordResetView.as_view()),
    # --------------------------------------------------------------------------------------
    # Override allauth `account_logout` URL.
    # /accounts/logout/ <=> account_logout
    # We need custom code to process PEAMU logout.
    re_path(r"^accounts/logout/$", dashboard_views.logout),
    # --------------------------------------------------------------------------------------
    # Other allauth URLs.
    path("accounts/", include("allauth.urls")),
    # --------------------------------------------------------------------------------------
    # PEAMU URLs.
    path("accounts/", include("itou.allauth_adapters.peamu.urls")),
    # FranceConnect URLs.
    path("franceconnect/", include("itou.france_connect.urls")),
    # --------------------------------------------------------------------------------------
    # API.
    path("api/v1/", include("itou.api.urls", namespace="v1")),
    path("api/", include("itou.www.api.urls")),
    # www.
    path("", include("itou.www.home.urls")),
    path("apply/", include("itou.www.apply.urls")),
    path("approvals/", include("itou.www.approvals_views.urls")),
    path("autocomplete/", include("itou.www.autocomplete.urls")),
    path("dashboard/", include("itou.www.dashboard.urls")),
    path("employee_record/", include("itou.www.employee_record_views.urls")),
    path("institutions/", include("itou.www.institutions_views.urls")),
    path("invitations/", include("itou.www.invitations_views.urls")),
    path("prescribers/", include("itou.www.prescribers_views.urls")),
    path("search/", include("itou.www.search.urls")),
    path("siae/", include("itou.www.siaes_views.urls")),
    path("signup/", include("itou.www.signup.urls")),
    path("stats/", include("itou.www.stats.urls")),
    path("versions/", include("itou.www.releases.urls")),
    # Enable Anymailâ€™s status tracking
    # https://anymail.readthedocs.io/en/stable/esps/mailjet/#status-tracking-webhooks
    re_path(r"^webhooks/anymail/", include("anymail.urls")),
    path("welcoming_tour/", include("itou.www.welcoming_tour.urls")),
    # Static pages.
    path("accessibility/", TemplateView.as_view(template_name="static/accessibility.html"), name="accessibility"),
    # Errors pages.
    path("403/", TemplateView.as_view(template_name="403.html"), name="403"),
    path("404/", TemplateView.as_view(template_name="404.html"), name="404"),
    path("500/", TemplateView.as_view(template_name="500.html"), name="500"),
]

if settings.DEBUG and "debug_toolbar" in settings.INSTALLED_APPS:
    import debug_toolbar

    urlpatterns = [path("__debug__/", include(debug_toolbar.urls))] + urlpatterns
    # In the local developpemnt environment, we enable the default callback URLs
    urlpatterns = [path("", include("itou.france_connect.urls"))] + urlpatterns
