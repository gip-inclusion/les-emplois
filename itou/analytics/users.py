from django.db.models import Count

from itou.users import enums as users_enums, models as users_models

from . import models


def collect_analytics_data(before):
    counts = {
        info["kind"]: info["count"]
        for info in users_models.User.objects.values("kind").annotate(count=Count("pk")).values("kind", "count")
    }
    return {
        models.DatumCode.USER_COUNT: sum(counts.values()),
        models.DatumCode.USER_JOB_SEEKER_COUNT: counts.get(users_enums.KIND_JOB_SEEKER, 0),
        models.DatumCode.USER_PRESCRIBER_COUNT: counts.get(users_enums.KIND_PRESCRIBER, 0),
        models.DatumCode.USER_EMPLOYER_COUNT: counts.get(users_enums.KIND_EMPLOYER, 0),
        models.DatumCode.USER_LABOR_INSPECTOR_COUNT: counts.get(users_enums.KIND_LABOR_INSPECTOR, 0),
        models.DatumCode.USER_ITOU_STAFF_COUNT: counts.get(users_enums.KIND_ITOU_STAFF, 0),
    }
