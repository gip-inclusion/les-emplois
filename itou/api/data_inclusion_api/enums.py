import enum


class StructureTypeStr(str, enum.Enum):
    ORGA = "orga"
    SIAE = "siae"
