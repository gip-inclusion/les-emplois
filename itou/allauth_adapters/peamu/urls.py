from django.urls import include, re_path

from itou.allauth_adapters.peamu import views as peamu_views
from itou.allauth_adapters.peamu.provider import PEAMUProvider


def default_urlpatterns(provider):
    urlpatterns = [
        re_path(r"^login/$", peamu_views.oauth2_login, name=provider.id + "_login"),
        re_path(r"^login/callback/$", peamu_views.oauth2_callback, name=provider.id + "_callback"),
    ]

    return [
        re_path(r"^profile/$", peamu_views.redirect_to_dashboard_view),
        re_path(r"^" + provider.get_slug() + r"/", include(urlpatterns)),
    ]


urlpatterns = default_urlpatterns(PEAMUProvider)
