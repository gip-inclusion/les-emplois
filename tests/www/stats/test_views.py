from datetime import datetime, timezone

import pytest
from django.test import override_settings
from django.urls import reverse
from freezegun import freeze_time
from pytest_django.asserts import assertQuerysetEqual

from itou.analytics.models import StatsDashboardVisit
from itou.common_apps.address.departments import DEPARTMENT_TO_REGION
from itou.institutions.enums import InstitutionKind
from itou.institutions.factories import InstitutionWithMembershipFactory
from itou.prescribers.factories import PrescriberOrganizationWithMembershipFactory
from itou.siaes.factories import SiaeFactory
from itou.users.factories import PrescriberFactory
from itou.utils.apis.metabase import METABASE_DASHBOARDS
from tests.utils.test import TestCase


class StatsViewTest(TestCase):
    @override_settings(METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar")
    def test_stats_public(self):
        url = reverse("stats:stats_public")
        response = self.client.get(url)
        assert response.status_code == 200

    def test_stats_pilotage_unauthorized_dashboard_id(self):
        url = reverse("stats:stats_pilotage", kwargs={"dashboard_id": 123})
        response = self.client.get(url)
        assert response.status_code == 403

    @override_settings(
        PILOTAGE_DASHBOARDS_WHITELIST=[123], METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar"
    )
    def test_stats_pilotage_authorized_dashboard_id(self):
        url = reverse("stats:stats_pilotage", kwargs={"dashboard_id": 123})
        response = self.client.get(url)
        assert response.status_code == 200

    @override_settings(
        PILOTAGE_DASHBOARDS_WHITELIST=[123], METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar"
    )
    def test_stats_pilotage_authorized_dashboard_id_while_authenticated(self):
        user = PrescriberFactory()
        self.client.force_login(user)
        url = reverse("stats:stats_pilotage", kwargs={"dashboard_id": 123})
        response = self.client.get(url)
        assert response.status_code == 200


def assert_stats_dashboard_equal(values):
    assertQuerysetEqual(
        StatsDashboardVisit.objects.all(),
        [
            values,
        ],
        transform=lambda visit: (
            visit.dashboard_id,
            visit.dashboard_name,
            visit.department,
            visit.region,
            visit.current_siae_id,
            visit.current_prescriber_organization_id,
            visit.current_institution_id,
            visit.user_kind,
            visit.user_id,
            visit.measured_at,
        ),
    )


@freeze_time("2023-03-10")
@override_settings(METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar")
@pytest.mark.parametrize(
    "view_name",
    [
        "stats_pe_delay_main",
        "stats_pe_delay_raw",
        "stats_pe_conversion_main",
        "stats_pe_conversion_raw",
        "stats_pe_state_main",
        "stats_pe_state_raw",
        "stats_pe_tension",
    ],
)
def test_stats_prescriber_log_visit(client, view_name):
    prescriber_org = PrescriberOrganizationWithMembershipFactory(authorized=True)
    user = prescriber_org.members.get()
    client.force_login(user)

    assertQuerysetEqual(StatsDashboardVisit.objects.all(), [])

    response = client.get(reverse(f"stats:{view_name}"))
    assert response.status_code == 200

    assert_stats_dashboard_equal(
        (
            METABASE_DASHBOARDS.get(view_name)["dashboard_id"],
            view_name,
            prescriber_org.department,
            DEPARTMENT_TO_REGION[prescriber_org.department],
            None,
            prescriber_org.pk,
            None,
            user.kind,
            user.pk,
            datetime(2023, 3, 10, tzinfo=timezone.utc),
        ),
    )


@freeze_time("2023-03-10")
@pytest.mark.parametrize(
    "view_name",
    [
        "stats_siae_etp",
        "stats_siae_hiring",
    ],
)
def test_stats_siae_log_visit(client, view_name, settings):
    siae = SiaeFactory(name="El garaje de la esperanza", with_membership=True)
    user = siae.members.get()

    settings.METABASE_SITE_URL = "http://metabase.fake"
    settings.METABASE_SECRET_KEY = "foobar"
    settings.STATS_SIAE_USER_PK_WHITELIST = [user.pk]

    client.force_login(user)

    assertQuerysetEqual(StatsDashboardVisit.objects.all(), [])

    response = client.get(reverse(f"stats:{view_name}"))
    assert response.status_code == 200

    assert_stats_dashboard_equal(
        (
            METABASE_DASHBOARDS.get(view_name)["dashboard_id"],
            view_name,
            siae.department,
            DEPARTMENT_TO_REGION[siae.department],
            siae.pk,
            None,
            None,
            user.kind,
            user.pk,
            datetime(2023, 3, 10, tzinfo=timezone.utc),
        ),
    )


@freeze_time("2023-03-10")
@override_settings(METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar")
@pytest.mark.parametrize(
    "view_name",
    [
        "stats_ddets_iae_auto_prescription",
        "stats_ddets_iae_follow_siae_evaluation",
        "stats_ddets_iae_iae",
        "stats_ddets_iae_siae_evaluation",
        "stats_ddets_iae_hiring",
    ],
)
def test_stats_ddets_iae_log_visit(client, view_name):
    institution = InstitutionWithMembershipFactory(kind=InstitutionKind.DDETS_IAE, department="22")
    user = institution.members.get()
    client.force_login(user)

    assertQuerysetEqual(StatsDashboardVisit.objects.all(), [])

    url = reverse(f"stats:{view_name}")
    response = client.get(url)
    assert response.status_code == 200
    # Check that the base slash of the URL is not included; it's added by the Javascript.
    assert response.context["matomo_custom_url"] == f"{url[1:]}/Bretagne/22---Cotes-d-Armor"

    assert_stats_dashboard_equal(
        (
            METABASE_DASHBOARDS.get(view_name)["dashboard_id"],
            view_name,
            institution.department,
            DEPARTMENT_TO_REGION[institution.department],
            None,
            None,
            institution.pk,
            user.kind,
            user.pk,
            datetime(2023, 3, 10, tzinfo=timezone.utc),
        ),
    )


@freeze_time("2023-03-10")
@override_settings(METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar")
@pytest.mark.parametrize(
    "view_name",
    [
        "stats_dreets_iae_auto_prescription",
        "stats_dreets_iae_follow_siae_evaluation",
        "stats_dreets_iae_iae",
        "stats_dreets_iae_hiring",
    ],
)
def test_stats_dreets_iae_log_visit(client, view_name):
    institution = InstitutionWithMembershipFactory(kind="DREETS IAE")
    user = institution.members.get()
    client.force_login(user)

    assertQuerysetEqual(StatsDashboardVisit.objects.all(), [])

    response = client.get(reverse(f"stats:{view_name}"))
    assert response.status_code == 200

    assert_stats_dashboard_equal(
        (
            METABASE_DASHBOARDS.get(view_name)["dashboard_id"],
            view_name,
            None,
            DEPARTMENT_TO_REGION[institution.department],
            None,
            None,
            institution.pk,
            user.kind,
            user.pk,
            datetime(2023, 3, 10, tzinfo=timezone.utc),
        ),
    )


@freeze_time("2023-03-10")
@override_settings(METABASE_SITE_URL="http://metabase.fake", METABASE_SECRET_KEY="foobar")
@pytest.mark.parametrize(
    "view_name",
    [
        "stats_dgefp_auto_prescription",
        "stats_dgefp_follow_siae_evaluation",
        "stats_dgefp_iae",
        "stats_dgefp_siae_evaluation",
        "stats_dgefp_af",
    ],
)
def test_stats_dgefp_log_visit(client, view_name):
    institution = InstitutionWithMembershipFactory(kind="DGEFP")
    user = institution.members.get()
    client.force_login(user)

    assertQuerysetEqual(StatsDashboardVisit.objects.all(), [])

    response = client.get(reverse(f"stats:{view_name}"))
    assert response.status_code == 200

    assert_stats_dashboard_equal(
        (
            METABASE_DASHBOARDS.get(view_name)["dashboard_id"],
            view_name,
            None,
            None,
            None,
            None,
            institution.pk,
            user.kind,
            user.pk,
            datetime(2023, 3, 10, tzinfo=timezone.utc),
        ),
    )
