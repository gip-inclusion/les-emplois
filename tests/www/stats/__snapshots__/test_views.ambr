# serializer version: 1
# name: test_webinar_banner_display
  "[{'title': 'Des questions sur l’utilisation des tableaux de bord ?', 'description': 'Nous y répondons lors d’un webinaire questions / réponses animé chaque mois.', 'call_to_action': 'Je m’inscris', 'url': 'https://app.livestorm.co/itou/le-pilotage-de-linclusion-professionnels-de-liae-questions-reponses-sur-les-tableaux-de-bord-1', 'is_displayable': True}]"
# ---
